### PR TITLE
dev/accessibility#10 : Moved datepicker widget to a separate file

### PR DESCRIPTION
**Pull request by monishdeb**
Overview
----------------------------------------
This is a subPR of https://github.com/civicrm/civicrm-core/pull/12580 and to move datepicker functionality to a separate file

### CODE REVIEW

File: js/Common.js [hunk 1]
```diff
@@ -646,155 +646,14 @@ if (!CRM.vars) CRM.vars = {};
...
   }
 
-  function copyAttributes($source, $target, attributes) {
+  CRM.utils.copyAttributes = function ($source, $target, attributes) {
     _.each(attributes, function(name) {
       if ($source.attr(name) !== undefined) {
         $target.attr(name, $source.attr(name));
       }
     });
   }
```
Comment (colemanw) [inline js/Common.js]: Jenkins says this line is missing a semicolon.

**Comment by eileenmcnaughton** [comment]
@monishdeb test errors relate

**Comment by monishdeb** [comment]
I see

@colemanw do I need to include the new file anywhere in test suite, as I checked on the UI where ```crm-mailing-radio-date``` directive is used and it's working correctly.

**Comment by colemanw** [comment]
Yes I suspect the new file needs to be included in Karma setup somewhere... but grepping around so far I can't find where.

**Comment by seamuslee001** [comment]
guessing maybe in this array https://github.com/civicrm/civicrm-core/blob/master/karma.conf.js#L25

**Comment by monishdeb** [comment]
Thanks, a lot @seamuslee001 indeed it seems to be the point.

**Comment by monishdeb** [comment]
Jenkins test this please

**Comment by seamuslee001** [comment]
Jenkins test this please

File: js/crm.datepicker.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+            val += (val ? ' ' : '') + $timeField.timeEntry('getTime').toTimeString().substr(0, 8);
+          }
+          $dataField.val(val).trigger('change', ['userInput']);
+        }
+      }
+      $dataField.hide().addClass('crm-hidden-date').on('change.crmDatepicker', updateInputFields);
+      updateInputFields();
+    });
+  };
+})(jQuery, CRM, CRM._);
```
Comment (seamuslee001) [inline js/crm.datepicker.js]: @monishdeb phantomjs is complaining about not able to find `CRM`  I think the date picker file needs to be loaded after Common.js

**Comment by seamuslee001** [comment]
Merging as per the tag

**Comment by monishdeb** [comment]
Thanks @seamuslee001 :)

**Pull request merged**